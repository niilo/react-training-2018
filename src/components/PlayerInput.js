import React, {Component} from 'react';
import PropTypes from 'prop-types';

class PlayerInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
    };
  }

  handleChange = event => {
    const value = event.target.value;
    this.setState(() => {
      return {
        username: value,
      };
    });
  };

  handleSubmit = event => {
    event.preventDefault();
    this.props.onSubmit(this.props.id, this.state.username);
  };

  render() {
    return (
      <form className="column" onSubmit={this.handleSubmit}>
        <label className="header" htmlFor="username">
          {this.props.label}
        </label>
        <input
          id="username"
          placeholder="github username"
          type="text"
          autoComplete="off"
          value={this.state.username}
          onChange={this.handleChange}
        />
        <button
          className="button"
          type="submit"
          disabled={!this.state.username}
        >
          Submit
        </button>
      </form>
    );
  }
}

PlayerInput.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

PlayerInput.defaultProps = {
  label: 'Username',
};

export default PlayerInput;
