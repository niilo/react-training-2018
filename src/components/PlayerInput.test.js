import React from 'react';
import renderer from 'react-test-renderer';
import PlayerInput from './PlayerInput';

it('renders default correctly', () => {
  const tree = renderer
    .create(
      <PlayerInput
        id="1"
        label="user1"
        onSubmit={function() {
          alert('hep');
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
