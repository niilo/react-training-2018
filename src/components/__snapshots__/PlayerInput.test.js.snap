// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default correctly 1`] = `
<form
  className="column"
  onSubmit={[Function]}
>
  <label
    className="header"
    htmlFor="username"
  >
    user1
  </label>
  <input
    autoComplete="off"
    id="username"
    onChange={[Function]}
    placeholder="github username"
    type="text"
    value=""
  />
  <button
    className="button"
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;
